agents/statsd/src/statsd_agent.cc

```diff
@@ -536,7 +559,10 @@ void StatsDAgent::do_start() {
   status(Initializing);
 
   if (hooks_init_ == false) {
-    ASSERT_EQ(0, OnConfigurationHook(config_agent_cb, weak_from_this()));
+    // Only add the OnConfigurationHook if this is the global instance.
+    if (Inst().get() == this) {
+      ASSERT_EQ(0, OnConfigurationHook(config_agent_cb, weak_from_this()));
+    }
```

Comment:
This is because of the test right? Why aren't you adding the ThreadAdded/RemovedHook methods inside that conditional as well?

Reply:
Mostly because I am being lazy. There are two main assumptions:
1. A `StatsDAgent` instance will be used for the entire process lifetime
2. At most 2 `StatsDAgent` instances will ever be created.

Using these assumptions, I decided it would be easier for each `StatsDAgent` instance to set up its own hooks when a new thread is created/destroyed rather than have a single global hook that would iterate through a list of `StatsDAgent` instances. This simplified the logic a lot and will be easy to fix if we ever add the ability to remove a hook.

Though, now that I type this up I realize that all the stuff I added for the `agent_list_` can be removed. Since it was never used. Will make that change.